docker/transport/npipesocket.py

```diff
@@ -131,22 +133,38 @@ def recv_into(self, buf, nbytes=0):
         if not isinstance(buf, memoryview):
             readbuf = memoryview(buf)
 
+        event = win32event.CreateEvent(None, True, True, None)
```

Comment:
This will leak the event handle (though it looks like `win32event` will close it as part of GC).

Can we do both of these with a `try` / `finally` block with a call to `win32api.CloseHandle(event)`?

Reply:
I added the try/finally block, tbh I'm not super familiar with the python windows api so I'm not sure if I added them at the right spots

Reply:
What you did looks great!

---

(On the Python side, `finally` will execute even if an exception gets thrown, so this ensures that the Windows API handle we create is released ASAP and even if we throw. On the Windows API side, lots of things use handles and it's good to not leak them 😅 )